### PR TITLE
upgrade notes entry for approle constraint and warning on role read

### PR DESCRIPTION
**Pull request by vishalnayak**

### CODE REVIEW

**Comment by LLBennett** [comment]
LGTM.

File: website/source/docs/install/upgrade-to-0.6.2.html.md [hunk 1]
```diff
@@ -10,3 +10,11 @@ description: |-
 
 This page contains the list of breaking changes for Vault 0.6.2. Please read it
 carefully.
+
+## AppRole Role Constraints
```
Comment (jefferai) [inline website/source/docs/install/upgrade-to-0.6.2.html.md]: Please don't forget to add this info to the changelog as well post-merge.

File: builtin/credential/approle/path_role.go [hunk 2]
```diff
@@ -521,8 +521,26 @@ func (b *backend) pathRoleSecretIDList(req *logical.Request, data *framework.Fie
 	return logical.ListResponse(listItems), nil
 }
 
-// setRoleEntry grabs a write lock and stores the options on an role into the storage.
-// Also creates a reverse index from the role's RoleID to the role itself.
+// validRoleConstraints tells if the role has at least one constraint enabled
+func validRoleConstraints(role *roleStorageEntry) (bool, error) {
```
Comment (jefferai) [inline builtin/credential/approle/path_role.go]: My main issue here is that it either always returns either `true, nil` or `false, <some error>` which means we only need one value. I think you should either remove the bool or the error, probably the bool, and it's simply valid if err is not nil. We can add in more complex logic later if needed.

File: website/source/docs/install/upgrade-to-0.6.2.html.md (hunk 1 above)
Comment (vishalnayak) [inline website/source/docs/install/upgrade-to-0.6.2.html.md]: CL is already updated.

Comment (jefferai) [inline website/source/docs/install/upgrade-to-0.6.2.html.md]: Oops, ok. Missed that!

File: builtin/credential/approle/path_role.go (hunk 2 above)
Comment (vishalnayak) [inline builtin/credential/approle/path_role.go]: It was that way, using just the error at first. I had to skip on checking role being nil in that case. I guess that's something we can live with. I'll revert it back to using just the error and punt on checking role for nil. And add a note in the method's comments.

Comment (jefferai) [inline builtin/credential/approle/path_role.go]: You could still check if the role is nil -- just return an error. Presumably this function shouldn't be getting called on a nil role anyways, so it seems fine to just error in that case (a nil role by definition does not have valid constraints, it seems to me).

Comment (vishalnayak) [inline builtin/credential/approle/path_role.go]: Makes sense to me. Done.

**Comment by jefferai** [comment]
🚢

**Pull request merged**